certora/specs/GeneralAdapter1Reverts.spec

```
@@ -173,20 +98,22 @@ rule revertOrStateChanged(env e, method f, calldataarg args) filtered {
          f.selector != sig:onMorphoFlashLoan(uint256, bytes).selector &&
          // Property checked in a different way.
          f.selector != sig:nativeTransfer(address, uint256).selector &&
-         f.selector != sig:wrapNative(uint256, address).selector &&
          f.selector != sig:unwrapNative(uint256, address).selector &&
```

For unwrapNative also the property doesn't hold. The best would be to either make it go through (so it's not in this filter list) or have a rule that only explains why it doesn't hold. So for this specific case of unwrapNative this would be:
```suggestion
         // Property doesn't hold for unwrapNative, see the rule wrapNativeChange.
         f.selector != sig:unwrapNative(uint256, address).selector &&
```
and then in `wrapNativeChange` only have the last assert:
```solidity
    // Check that state doesnt change when using amount equals zero.
    assert amount == 0 => receiverWrappedBalanceBefore == WETH.balanceOf(receiver) && adapterNativeBalanceBefore == nativeBalances[currentContract];
```
as it's the one showing that for some input the state doesn't change. Or even simpler, just
```solidity
    assert amount == 0 => storageBefore == lastStorage;
```

---

Some thing like in https://github.com/morpho-org/bundler3/pull/229/commits/9e82d7da6c09d549db5793122c2a5aed42475a6d?

---

Exactly, perfect :100:

---

Done in https://github.com/morpho-org/bundler3/pull/229/commits/5fe46e1bf67f37a9606ed2d94eac48882753a4f4.